Source/FakeItEasy/Creation/DefaultFakeCreatorFacade.cs

```diff
@@ -126,6 +126,11 @@ public IFakeOptionsBuilder<T> Implements(Type interfaceType)
                 return this;
             }
 
+            public IFakeOptionsBuilder<T> Implements<TInterface>()
+            {
+                return Implements(typeof (TInterface));

```

Comment:
Oh, I see. I don’t have StyleCop, which explains why I wasn’t getting the warnings

 

From: Blair Conrad [mailto:notifications@github.com] 
Sent: dimanche 29 mars 2015 03:03
To: FakeItEasy/FakeItEasy
Cc: Thomas Levesque
Subject: Re: [FakeItEasy] Add generic overload of Implements (#471)

 

In Source/FakeItEasy/Creation/DefaultFakeCreatorFacade.cs <https://github.com/FakeItEasy/FakeItEasy/pull/471#discussion_r27350332> :

> @@ -126,6 +126,11 @@ public IFakeOptionsBuilder<T> Implements(Type interfaceType)
>                  return this;
>              }
>  
> +            public IFakeOptionsBuilder<T> Implements<TInterface>()
> +            {
> +                return Implements(typeof (TInterface));

Sorry, technically StyleCop warnings. I am not sophisticated, and am sometimes linguistically lazy, so called them "compiler warnings".

I see them in the IDE and the rake build:

 <https://cloud.githubusercontent.com/assets/3275797/6883641/475b2756-d58d-11e4-968e-8988178985e1.png> 

 <https://cloud.githubusercontent.com/assets/3275797/6883647/6b0936ca-d58d-11e4-8140-48f613f269ec.png> 

And as for ReSharper, I do have 

<s:Boolean x:Key="/Default/CodeStyle/CodeFormatting/CSharpFormat/SPACE_BEFORE_TYPEOF_PARENTHESES/@EntryValue">False</s:Boolean>

in my FakeItEasy.sln.DotSettings.user file, so that probably explains the difference in behaviour.

—
Reply to this email directly or view it on GitHub <https://github.com/FakeItEasy/FakeItEasy/pull/471/files#r27350332> .  <https://github.com/notifications/beacon/ABxyWVquKFJi8FxKHBeumftwqZDZV0MRks5n50bNgaJpZM4D16jk.gif>